full-node/db/sov-schema-db/src/snapshot.rs

```diff
@@ -104,6 +115,129 @@ impl<Q: QueryManager> DbSnapshot<Q> {
             .expect("SchemaBatch lock must not be poisoned")
             .delete(key)
     }
+
+    /// Get value of largest key written value for given [`Schema`]
+    pub fn get_largest<S: Schema>(&self) -> anyhow::Result<Option<S::Value>> {
+        let local_cache = self
+            .cache
+            .lock()
+            .expect("SchemaBatch lock must not be poisoned");
+        let local_cache_iter = local_cache.iter::<S>();
+
+        let parent = self
+            .parents_manager
+            .read()
+            .expect("Parent lock must not be poisoned");
+
+        let parent_iter = parent.iter::<S>(self.id)?;
+
+        let mut combined_iter: SnapshotIter<'_, Q, S> = SnapshotIter {
+            local_cache_iter: local_cache_iter.peekable(),
+            parent_iter: parent_iter.peekable(),
+        };
+
+        if let Some((_, value)) = combined_iter.next() {
+            let value = S::Value::decode_value(&value)?;
+            return Ok(Some(value));
+        }
+
+        Ok(None)
+    }
+
+    /// Get value in [`Schema`] that is smaller or equal than give `seek_key`
+    pub fn get_prev<S: Schema>(
+        &self,
+        seek_key: &impl SeekKeyEncoder<S>,
+    ) -> anyhow::Result<Option<S::Value>> {
+        let seek_key = seek_key.encode_seek_key()?;
+
+        let local_cache = self
+            .cache
+            .lock()
+            .expect("Local cache lock must not be poisoned");
+
+        let local_cache_iter = local_cache.iter::<S>();
+
+        let parent = self
+            .parents_manager
+            .read()
+            .expect("Parent snapshots lock must not be poisoned");
+
+        let parent_iter = parent.iter::<S>(self.id)?;
+
+        let combined_iter: SnapshotIter<'_, Q, S> = SnapshotIter {
+            local_cache_iter: local_cache_iter.peekable(),
+            parent_iter: parent_iter.peekable(),
+        };
+
+        if let Some((_, value)) = combined_iter
+            .filter_map(|(key, value)| {
+                if key <= seek_key {
+                    return Some((key, value));
+                }
+                None
+            })
+            .next()
+        {
+            return Ok(Some(S::Value::decode_value(&value)?));
+        }
+
+        Ok(None)
+    }
+}
+
+struct SnapshotIter<'a, Q: QueryManager + 'a, S: Schema> {
+    local_cache_iter: std::iter::Peekable<SchemaBatchIterator<'a, S>>,
+    parent_iter: std::iter::Peekable<Q::Iter<'a, S>>,
+}
+
+impl<'a, Q: QueryManager + 'a, S: Schema> Iterator for SnapshotIter<'a, Q, S> {
+    type Item = (SchemaKey, SchemaValue);
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
This look squite complex, do we have a unit test for it?

Reply:
Not directly. Intergration tests `tests/iterator_test` cover that via `get_last` and `get_prev`

But I can add additional iterator test just for it

Reply:
Let's add them

Reply:
Done!